Objects/interpreteridobject.c

```diff
@@ -85,8 +53,31 @@ interpid_new(PyTypeObject *cls, PyObject *args, PyObject *kwds)
         id = ((interpid *)idobj)->id;
     }
     else {
-        id = _Py_CoerceID(idobj);
+        PyObject *pyid;
+        if (PyIndex_Check(idobj)) {
+            pyid = idobj;
+            Py_INCREF(pyid);
+        }
+        else if (PyUnicode_Check(idobj)) {
```

Comment:
I am not sure strings should be supported in constructor. The error message says that the argument must be an int. And the code can be simpler without supporting strings. What is the use case for this?